backend/src/test/java/asegroup1/api/services/landregistry/LandRegistryServiceImplTest.java

```diff
@@ -188,45 +190,211 @@ void testIfInvalidPostCodeFetchesNullCoordinates() {
+        assert heatMapDataPoints.get(0).getColour().getRed() < heatMapDataPoints.get(2).getColour().getRed()
+                && heatMapDataPoints.get(2).getColour().getRed() < heatMapDataPoints.get(1).getColour().getRed();
+
+    }
+
+    @Test
+    void testIfCorrectLandRegistryDataIsFetchedForPostcode() {
+        Random random = new Random(RANDOM_SEED);
+
+        List<String> postcodes = Collections.singletonList("BN11 4AA");
+
+        List<LandRegistryData> postCodeLocationData = new ArrayList<>();
+
+        for (String postcode1 : postcodes) {
+            LandRegistryData landRegistryData = new LandRegistryData();
+            landRegistryData.setPostCode(postcode1);
+            landRegistryData.setLatitude(random.nextDouble());
+            landRegistryData.setLatitude(random.nextDouble());
+
+            postCodeLocationData.add(landRegistryData);
+        }
+
+        List<LandRegistryData> postcodesFetchedMock = new ArrayList<>();
+
+        for (String postcode : postcodes) {
+            LandRegistryData landRegistryData = new LandRegistryData();
+            landRegistryData.setPostCode(postcode);
+            postcodesFetchedMock.add(landRegistryData);
+        }
+
+        JSONObject mockRequest = new JSONObject();
+
+        try {
+            mockRequest.put("top", 50.814);
+            mockRequest.put("right", -0.376);
+            mockRequest.put("bottom", 50.8135);
+            mockRequest.put("left", -0.378);
+
+
+            LandRegistryDaoImpl landRegistryDataDaoMock = mock(LandRegistryDaoImpl.class);
+            when(landRegistryDataDaoMock.searchForLandRegistryDataInBoundaries(
+                    mockRequest.getDouble("top"),
+                    mockRequest.getDouble("right"),
+                    mockRequest.getDouble("bottom"),
+                    mockRequest.getDouble("left")
+            )).thenReturn(postcodesFetchedMock);
+
+            when(landRegistryDataDaoMock.searchForLandRegistryDataInBoundaries(
+                    mockRequest.getDouble("top"),
+                    mockRequest.getDouble("right"),
+                    mockRequest.getDouble("bottom"),
+                    mockRequest.getDouble("left")
+            )).thenReturn(postCodeLocationData);
+
+            landRegistryService = new LandRegistryServiceImpl(landRegistryDataDaoMock);
+
+            assert landRegistryService.getPositionInsideBounds(mockRequest).size() == 1;
+        } catch (Exception e) {
+            e.printStackTrace();
+            assert false;
+        }
+    }
+
+    @Test
+    void testIfPostcodesAreAggregatedCorrectly() {
+        LandRegistryDaoImpl landRegistryDataDaoMock = mock(LandRegistryDaoImpl.class);
+
+        String[] postcodes = {"BN14 7BH", "NW9 9PR", "NN12 8DT", "TW7 4QN", "L22 3YU", "RM17 6LJ", "RG14 7DF", "SE25 5RT"};
+
+
+        List<LandRegistryData> landRegistryDataList = new ArrayList<>();
+
+        for (int i = 0; i < LandRegistryServiceImpl.getAggregationLevels()[1] + 1; i++) {
```

Comment:
instead of adding one to getAggregationLevels you could change the operation to <=